### PR TITLE
Refine change password form to provide current password

### LINKED ISSUE #12: Improve change password form in the settings
# Feature improvement

When user wants to change the password, user only input new password and confirm for new password. I think user must input the previous or current password for safety.

The solution or improvement is add current password field in change password page. So the user who wants to change the password must input the current password.

### PR DESCRIPTION
**Pull request by fikrialwan**
## What kind of change does this PR introduce?

fix #12 
Improve change password page to add current password field. 

## What is the current behavior?

The current behavior is user only input new password and confirm new password
image

## What is the new behavior?

The new behavior is user must input current password too for change the password
image

when user input the wrong current password
image

### CODE REVIEW

**Comment by mhaidarhanif** [comment]
Thanks @fikrialwan 

@bearmentor/collaborator We'll be reviewing this

**Review by zainfathoni — APPROVED** [review]

File: app/models/user-password.server.ts [hunk 1]
```diff
@@ -13,9 +13,29 @@ export const query = {
 }
 
 export const mutation = {
-  async update({ id, password }: z.infer<typeof schemaUserUpdatePassword>) {
+  async update({
+    id,
+    password,
+    currentPassword,
+  }: z.infer<typeof schemaUserUpdatePassword>) {
     const hashedPassword = await bcrypt.hash(password, 10)
```
Comment (r17x) [inline app/models/user-password.server.ts:21]:
```suggestion
    const hashedPassword = await bcrypt.hash(password, 10)
```

I think you can be move in line *38*.

**Review by r17x — CHANGES_REQUESTED** [review]

**Review by r17x — APPROVED** [review]
Looks good to me. nice update @fikrialwan

File: app/models/user-password.server.ts (hunk 1 above)
Comment (fikrialwan) [reply app/models/user-password.server.ts]: Hi @r17x , I'm Fikri and Nice to meet you too. I'm new comer contributor too and this is my first pull request, thank you so much for the feedback.

**Comment by mhaidarhanif** [comment]
@fikrialwan Looks nice 👍

But can we improve it a bit? After we submit the password, the form inputs should be cleared. And maybe show a toast message.

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
@fikrialwan Looks nice 👍 I've tested the functionality.

But can we improve it a bit? After we submit the password, the form inputs should be cleared. This can be done in the same branch/PR.

Then maybe show a toast message after it's done.

**Review by mhaidarhanif — APPROVED** [review]

**Comment by mhaidarhanif** [comment]
Thanks @fikrialwan this improvement has been merged.

cc @bearmentor/collaborator

**Pull request merged**